src/smos/smos_l4/interface.py

```diff
@@ -0,0 +1,598 @@
+# -*- coding: utf-8 -*-
+
+# The MIT License (MIT)
+#
+# Copyright (c) 2021, TU Wien
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import warnings
+import numpy as np
+import os
+from pygeobase.io_base import ImageBase, MultiTemporalImageBase
+from pygeobase.object_base import Image
+from pynetcf.time_series import GriddedNcOrthoMultiTs
+from datetime import timedelta, datetime
+from netCDF4 import Dataset, date2num, num2date
+from pygeogrids.netcdf import load_grid
+from smos.grid import EASE25CellGrid
+from smos.smos_ic import dist_name as subdist_name
+from smos import dist_name, __version__
+from collections import OrderedDict
+import pandas as pd
+
+
+class SMOSL4Img(ImageBase):
```

Comment:
a lot of duplicate code here as well, maybe we should make a common parent class that the smos-ic and l4 `Image` and `Dataset` classes then inherit from? And adapt it according to each dataset. I think that would make the whole package a bit more slim and easier to maintain, resp add additional products.

Reply:
agree. other review requests I completed I believe. 
merge and make this and issue?